ts/components/session/SessionIcon.tsx

```diff
@@ -0,0 +1,240 @@
+    viewBox: '-1 -2 15 15',
+  },
+  Star: {
+    path:
+      'M9.80779568,8.70262392 C9.66225594,8.99747141 9.38107073,9.20193068 9.05571654,9.24948607 L4.1495,9.9666031 L7.69882113,13.4236419 C7.93469487,13.6533829 8.0423575,13.9845141 7.98669695,14.3090433 L7.14926913,19.1916734 L11.5356371,16.8849265 C11.8270199,16.7316912 12.1751567,16.7316912 12.4665396,16.8849265 L16.8529075,19.1916734 L16.0154797,14.3090433 C15.9598192,13.9845141 16.0674818,13.6533829 16.3033555,13.4236419 L19.8526767,9.9666031 L14.9464601,9.24948607 C14.6211059,9.20193068 14.3399207,8.99747141 14.194381,8.70262392 L12.0010883,4.25925434 L9.80779568,8.70262392 Z M8.24682697,7.3464661 L11.104381,1.55737608 C11.4712164,0.814207972 12.5309603,0.814207972 12.8977957,1.55737608 L15.7553497,7.3464661 L22.1457165,8.28051393 C22.9656312,8.40035674 23.2924147,9.40819801 22.6988211,9.98635811 L18.0756101,14.4893656 L19.166697,20.8509567 C19.3068155,21.6679189 18.4492666,22.2908819 17.7156371,21.9050735 L12.0010883,18.8998497 L6.28653961,21.9050735 C5.55291004,22.2908819 4.69536119,21.6679189 4.83547972,20.8509567 L5.92656655,14.4893656 L1.30335554,9.98635811 C0.709762006,9.40819801 1.03654545,8.40035674 1.85646012,8.28051393 L8.24682697,7.3464661 lZ',
+    viewBox: '0 0 22 21',
+  },
+};
+
+export enum SessionIconSizes {
+  Small = 'small',
+  Medium = 'medium',
+  Large = 'large',
+}
+
+export interface Props {
+  iconType: SessionIconTypes;
+  iconSize: SessionIconSizes;
+  iconColor: string;
+  iconRotation: number;
+  onClick: any;
+}
+
+export class SessionIcon extends React.PureComponent<Props> {
+  public static defaultProps = {
+    iconSize: SessionIconSizes.Medium,
+    iconColor: '',
+    iconRotation: 0,
+  };
+
+  constructor(props: any) {
+    super(props);
+  }
+
+  public render() {
+    const { iconType, iconSize, iconColor, iconRotation } = this.props;
+
+    let iconDimensions;
+    switch (iconSize) {
+      case SessionIconSizes.Small:
+        iconDimensions = '15';
+        break;
+      case SessionIconSizes.Medium:
+        iconDimensions = '20';
+        break;
+      case SessionIconSizes.Large:
+        iconDimensions = '30';
+        break;
+      default:
+        iconDimensions = '20';
+    }
+
+    let iconDef = { path: '', viewBox: '' };
```

Comment:
These lines causes an error on the build:
`ERROR: 143:3  cyclomatic-complexity  The function render has a cyclomatic complexity of 22 which is higher than the threshold of 20`

Reply:
An idea would be to be able to do something like:
`iconDef = icons[iconType]`

Reply:
Yep it would be possible if you set the keys in `icons` to `SessionIconTypes.<iconType>`
E.g `Arrow` -> `SessionIconTypes.Arrow`.

We can also rename `SessionIconTypes` to `Types` since it will be namespaced under `SessionIcon` anyway